explorer/interpreter/value.h

```diff
@@ -121,6 +121,13 @@ class Value {
   const Kind kind_;
 };
 
+// Returns whether the fully-resolved kind that this value will eventually have
+// is currently unknown, because it depends on a generic parameter.
+inline bool IsValueKindDependent(Nonnull<const Value*> type) {
```

Comment:
FWIW I think this name a bit hard to interpret, maybe "IsDependentType" similar to "IsConcreteType" or "IsType"?

Reply:
Hm, this doesn't tell you whether the type is dependent, at least in the C++ sense. For example, `Vector(T)` would probably be considered a dependent type, but the value kind (`NominalClassType`) is not dependent -- this is specifically determining whether the `Value::Kind` is dependent. I'm having a hard time finding a better name that's still appropriate.